modules/integration/tests-common/admin-clients/src/main/java/org/wso2/bps/integration/common/clients/bpmn/ActivitiRestClient.java

```diff
@@ -0,0 +1,736 @@
+		}
+		catch(JSONException jsonMessage){
+			log.error("JSON Conversion Failure",jsonMessage);
+		}
+		return new String[] { null, null, null };
+	}
+
+
+	/**
+	 * Method is used to acquire deployment details using the deployment ID
+	 *
+	 * @param ID Deployment ID of the BPMN Package
+	 * @returns String Array with status, deploymentID and Name
+	 * @throws java.io.IOException
+	 * @throws org.json.JSONException
+	 */
+	public String[] getDeploymentById(String ID) throws IOException,JSONException {
+
+		String url = serviceURL+"repository/deployments/"
+		             +ID;
+		HttpHost target = new HttpHost(hostname,port,"http");
+		
+		DefaultHttpClient httpClient = new DefaultHttpClient();
+		httpClient.getCredentialsProvider().setCredentials
+				(new AuthScope(target.getHostName(),target.getPort()),
+		         new UsernamePasswordCredentials(username,password));
+
+		try {
+			HttpGet httpget = new HttpGet(url);
+			HttpResponse response = httpClient.execute(httpget);
+			String status = response.getStatusLine().toString();
+			String responseData = EntityUtils.toString(response.getEntity());
+			JSONObject jsonResponseObject = new JSONObject(responseData);
+			if (status.contains("201") || status.contains("200")) {
+				String deploymentID = jsonResponseObject.getString("id");
+				String name = jsonResponseObject.getString("name");
+				return new String[] { status, deploymentID, name };
+			}
+		}
+		catch (IOException ioMessage){
+			log.error("Get Request Failure",ioMessage);
+		}
+		catch (JSONException jsonMessage){
+			log.error("JSON Conversion Failure",jsonMessage);
+		}
+		return new String[] { null, null, null };
+	}
+
+	/**
+	 * Method is used to undeploy/remove a deployment from the server
+	 *
+	 * @param deploymentID used to identify the deployment to be removed
+	 * @return String with the Status
+	 * @throws IOException
+	 */
+
+	public String unDeployPackage(String deploymentID) throws IOException{
+
+		String url = serviceURL+"repository/deployments/"
+		             + deploymentID;
+		HttpHost target = new HttpHost(hostname,port,"http");
+		DefaultHttpClient httpClient = new DefaultHttpClient();
+		httpClient.getCredentialsProvider().setCredentials
+				(new AuthScope(target.getHostName(),target.getPort()),
+		         new UsernamePasswordCredentials(username,password));
+		try {
+			HttpDelete httpDelete = new HttpDelete(url);
+			HttpResponse response = httpClient.execute(httpDelete);
+			return response.getStatusLine().toString();
+		}
+		catch (IOException ioMessage){
+			log.error("Delete Request Failure",ioMessage);
+		}
+		return null;
+	}
+	//endregion
+
+		//region Process Definition Methods
+
+	/**
+	 * Method to find the definitionID which is necessary to start a process instance
+	 *
+	 * @param deploymentID the deployment id is used to identify the deployment uniquely
+	 * @return String Array containing status and definitionID
+	 * @throws IOException
+	 * @throws JSONException
+	 */
```

Comment:
Lets rename this method to findProcessDefinitionInfoById,  tells what the method is doing